tests/suites/test_suite_psa_crypto.function

```diff
@@ -5321,6 +5314,26 @@ exit:
 }
 /* END_CASE */
 
+/* BEGIN_CASE */
+void derive_large_key( int alg_arg,
+                       data_t *key_data, data_t *input1, data_t *input2,
+                       int type_arg, int bits_arg,
+                       int expected_status_arg )
+{
+    size_t key_bytes = PSA_BITS_TO_BYTES(bits_arg);
+    uint8_t* buffer = NULL;
+
+    /* Check that the target running this test can accomodate large
+     * keys on its heap, before calling the actual generate_key test */
+    ASSERT_ALLOC_WEAK(buffer, key_bytes);
```

Comment:
This is fragile and I fear that we'll run into problems in practice if the heap just barely has room for the key, but not for the key plus other data that the test allocates. Then the buffer allocation would succeed here, but the test would run out of memory later.

I would prefer to go through the test, and in the large-key case, allow the function that's supposed to allocate a large amount of memory to return `PSA_ERROR_OUT_OF_MEMORY.

Reply:
Actually there's a case where this definitely won't work: on a target where crypto runs in a separate service, this checks whether the client has room on its heap, but what matters for the test to pass is that the service has enough room. So please change to actually running the test, and in the “large” case, allowing `PSA_ERROR_OUT_OF_MEMORY`.

This is also better because it gives us a chance to catch errors in out-of-memory cleanup.

Reply:
Done.